### ISSUE TITLE
Edit current highlighted feed url

### ISSUE THREAD

**Issue by cannyboy** [newsboat/newsboat#1262]
I often want to edit (or delete) the feed that is currently highlighted. Since I have hundred of feeds, this can get a bit painful. I have to scroll down or search for the url after pressing E to edit.

What would be useful is (by pressing lowercase ‘e’) it could go directly to the line in the text editor.  In vim for instance, this is achieved by:

`vim +99 urls`

... to go to the 99th line of urls. Then I could edit and jump back quickly without much hassle.

**Comment by Minoru** [comment]
Yup, good idea. Previous discussion in Newsbeuter: https://github.com/akrennmair/newsbeuter/issues/258

I think the main obstacle here is that different editors (might) use different CLI options for this. Since the editor in Newsboat is configured via the standard `VISUAL` and `EDITOR` environment variables, we can't expect the user to add these specifics to the environment. I see two ways out:

1. **teach Newsboat about the different editors and their CLI options**. Editor detection might be brittle, and I'm not looking forward to maintaining that list;

2. **introduce a new setting in Newsboat which will include that info**. Ugh, more settings. Also, need to come up with the syntax. Would really prefer for this to work "automagically".

Can anyone think of other solutions? Are anyone aware of programs that already implement this?

**Comment by cannyboy** [comment]
Just did a little testing and vim, neovim, nano, emacs all use the `+99` syntax for jumping to the line number. I'm guessing that covers over 90% of users? Perhaps it's possible for newsboat to just disable the line jump shortcut if they don't use these common editors.

**Comment by Minoru** [comment]
Ah, cool! I somehow assumed that Emacs would have a very different way of doing this, but I'm glad to be wrong.

On to the detection, then. I think just taking the final component of the path is fine, e.g. `/usr/bin/vim` means "vim". I can think of a couple cases there this will break:

1. /usr/bin/vim.basic, /usr/bin/vim.tiny won't be detected as "vim", and won't be supported (Debian packages Vim in multiple configurations, with different suffixes to avoid conflicts);
2. /usr/bin/editor won't be detected as anything, and won't be supported (again, on Debian, it's a symlink to /etc/altlernatives/editor, which is itself a symlink to an actual executable).

I think we can go with a simple approach for now, and make it more intelligent as need arise.

@cannyboy, wanna take a stab at implementing this? :)

**Comment by m040601** [comment]
> Can anyone think of other solutions? Are anyone aware of programs that already implement this?

I can't think of it right away. But I will sure do my best to do my searches on this topic, now that I see there is interest and motivation in this.

What I can say, as a long time (years and years) Newsboat use is that this one , is one of those tiny little things that make a huge impact.
Totally worth the effort put into it.

> 
..... vim, neovim, nano, emacs ....I'm guessing that covers **_over 90% of users_**? ....

This. I like this line of reasoning. Assuming these kind of functionaliy will never be 100% satisfactory for 100% of all users, relentellessly focusing on the main "customers". 


> ...to the detection, then.....think of a couple cases there this will break...

Perhaps, once this preliminar implementation gains a little bit of steam, and in order to gain more feedback from end users, and test edge cases, publicize it more. Make it known that Newsboat is working on it. 

I am sure that a great number of long time power users, dont follow this github issues queue or the mailing list. And since Newsboat doesnt have a user wiki, do post it on Reddit or similar.

Something like "What's cooking for Release XYZ"  or "Status update Newsboat future plans" etc.

It would be really nice to publicize better what has been an excelent year in Newsboat development.

**Comment by Minoru** [comment]
> This. I like this line of reasoning. Assuming these kind of functionaliy will never be 100% satisfactory for 100% of all users, relentellessly focusing on the main "customers".

In case I came across as a perfectionist chasing clouds: my goal when asking for the best possible solution is not to dissuade from working on this, but to explore possibilities and get a detailed picture of the strengths and weaknesses of a particular solution. User-visible changes are held to an even higher standard because they are hard to fix afterwards; by exploring alternatives, we also find out how extensible each solution is.

> Something like "What's cooking for Release XYZ" or "Status update Newsboat future plans" etc.

Contributors to this repo are all volunteers, and development happens when someone feels like doing something. So the only "global" plan I have is to make a release every three months. Apart from that, this project is driven entirely by contributors' inclinations.

> It would be really nice to publicize better what has been an excelent year in Newsboat development.

I already write [the changelog](https://github.com/newsboat/newsboat/blob/6157052aa6c8f999b21d502d4b605574b8884529/CHANGELOG.md), [some reflections](https://groups.google.com/forum/#!topic/newsboat/LMKbAJuYeX0) for releases, and [blog posts](https://blog.debiania.in.ua/tags/newsboat-rust-rewrite.html) on some technical challenges. These are the kinds of things I like to *read*, so that's what I write as well. I personally don't read many "highlights posts", so I wouldn't be the best person to write one, either.

Luckily, this is where the community picks up the slack. I've seen at least a dozen YouTube videos about Newsboat, and read a couple blog posts about particular setups. So, just like with Newsboat's development itself: if you want to see some sort of coverage, the best thing to do is to start producing it ;)

**Comment by der-lyse** [comment]
Great idea, I have exactly the same use-case and would greatly appreciate such a feature!

Here's a potential pitfall for the implementation: I'm not sure whether Newsboat already remembers the original line number for each feed URL. This is important as empty lines and commented out lines are ignored when loading the _urls_ file. Therefore simply using the feed index won't work as users might end up at a wrong line. This might also be the case if feeds are differently sorted than in the defined file (but I haven't checked that code either). I reckon starting a wrong and "random" line is even more annoying than having to scroll to the correct position right from the very top.

Maybe the `edit-urls` operation can be just extended to open the _urls_ file at the appropriate line number if a supported editor is used and no new operation is needed for that. The article view does not support `E` (`edit-urls`) by default at the moment, however `e` is already bound to the `enqueue` operation by default. `E` could be used in all three views – feed list, article list and article view – to edit the _urls_ file at the line of the currently selected feed.

Since I'm a vim user, going with the simple approach first sounds good to me. ;-)

**Comment by Minoru** [comment]
> Here's a potential pitfall for the implementation: …

Ah, right. Also, feeds received from a remote API like Inoreader won't have a line number at all, because they aren't in the urls file.

> Maybe the edit-urls operation can be just extended to open the urls file at the appropriate line number

Definitely. I overlooked the `e`/`E` spelling in the top post, and just assumed that this is extending the `edit-urls` op instead of introducing a new one.

**Comment by dennisschagt** [comment]
We have just merged a change which makes this possible (https://github.com/newsboat/newsboat/pull/3339)
Example config: `bind e feedlist,articlelist,searchresultslist edit-urls "vim +%L %f"`

Docs:
<img width="1020" height="267" alt="Image" src="https://github.com/user-attachments/assets/c226f01e-f617-4f20-af5a-4d35bbd59589" />

**Comment by Minoru** [comment]
I think that change resolves the issue. If there's still some functionality missing, please open a new one.
